.circleci/config.yml

```diff
@@ -0,0 +1,596 @@
+
+jobs:
+  compile-only-job:
+    executor:
+      name: build-executor
+    working_directory: *pepper_apis_path
+    steps:
+      - checkout:
+          path: *repo_path
+      - setup-shared-env
+      - compile
+      # will be needed for tests later on
+      - render-configs
+      - persist_to_workspace:
+          root: *root_path
+          paths:
+            - repo
+
+  route-tests-job:
+    executor:
+      name: test-executor
+    working_directory: *pepper_apis_path
+    steps:
+      - setup-test-workspace
+      - run-maven-with-pepper-options:
+          phase: test
+          additional_options: '-Dtest=IntegrationTestSuite'
+
+
+  parallel-tests-job:
+    executor:
+      name: test-executor
+    working_directory: *pepper_apis_path
+    parallelism: <<parameters.parallelism>>
+    parameters:
+      parallelism:
+        default: 2
+        type: integer
+      repo_path:
+        type: string
+        default: *repo_path
+      circleci_path:
+        type: string
+        default: *circleci_path
+      m2_repo:
+        type: string
+        default: *m2_repo_path
+    steps:
+      - setup-test-workspace
+      - initialize-databases
+      - run:
+          name: Split tests
+          command: |
+            set +x
+            set +e
+            set -o xtrace
+            TEST_DATA_PATH='<< parameters.circleci_path >>/tests'
+            mkdir -p "$TEST_DATA_PATH"
+
+            TEST_SRC_DIRECTORY='src/test/java'
+            TEST_SEARCH_PATTERN='**/*Test.java'
+
+            SUREFIRE_CLASS_NAMES_PATH="${TEST_DATA_PATH}/surefire_classnames"
+
+            # generate list of tests to be processed
+            circleci tests glob "${TEST_SRC_DIRECTORY}/${TEST_SEARCH_PATTERN}" | # fancy circleci find command
+              sed -e "s#^${TEST_SRC_DIRECTORY}/\(.*\)\.java#\1#" | # Strip .java and parent directory path
+              tr "/" "." |  # Convert to a class name of form org.package.ClassName
+              grep -v '.*RouteTest' > "${SUREFIRE_CLASS_NAMES_PATH}" #Exclude RouteTest. These done from suite
+
+            # Test file name pattern does not pick this one up. Add explicitly
+            # echo 'org.broadinstitute.ddp.route.IntegrationTestSuite'  >> $SUREFIRE_CLASS_NAMES_PATH
+
+            echo "Total number of test class names collected: $(wc -l $SUREFIRE_CLASS_NAMES_PATH)"
+
+            # tests split will figure out which tests to run in this node
+            THIS_NODE_TESTS_PATH=/tmp/this_node_tests
+            cat $SUREFIRE_CLASS_NAMES_PATH | \
+              circleci tests split --split-by=timings --timings-type=classname > $THIS_NODE_TESTS_PATH
+
+
+            IGNORE_CLASS_LIST_PATH="${TEST_DATA_PATH}/surefire_classnames_ignore_list"
+
+            # Filter out the ones that don't run in this node
+            cat $SUREFIRE_CLASS_NAMES_PATH | \
+              grep -xvf $THIS_NODE_TESTS_PATH > $IGNORE_CLASS_LIST_PATH
+
+            // save path for next steps
```

Comment:
```suggestion
            # save path for next steps
```

Reply:
I tripped over this one before I saw your comment.